### PR TITLE
test: check tech debt for old nodejs versions

### PR DESCRIPTION
**Pull request by justinmk3**
## Problem

some code is using workarounds for older nodejs versions.

## Solution

- add a test that fails when the minimum nodejs version is 16+
- show nodejs version in the `About Toolkit` command

image

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/shared/vscode/env.ts [hunk 1]
```diff
@@ -88,3 +88,10 @@ export function isMinimumVersion(): boolean {
 export function getMinVscodeVersion(): string {
     return packageJson.engines.vscode.replace(/[^~]/, '')
 }
+
+/**
+ * Returns the minimum nodejs version declared in `package.json`.
+ */
+export function getMinNodejsVersion(): string {
+    return packageJson.devDependencies['@types/node'].replace(/[^~]/, '')
```
Comment (justinmk3) [inline src/shared/vscode/env.ts:96]: not ideal, but this informally (now formally) indicates what nodejs version we are targeting

Comment (JadenSimon) [reply src/shared/vscode/env.ts:96]: You can add `node` to the `engines` section.

File: src/test/techdebt.test.ts [hunk 2]
```diff
@@ -23,4 +23,13 @@ describe('tech debt', function () {
...
     })
+
+    it('nodejs minimum version', async function () {
+        const minNodejs = env.getMinNodejsVersion()
+
+        assert.ok(
+            semver.lt(minNodejs, '16.0.0'),
+            'remove require("perf_hooks").performance workarounds, use globalThis.performance instead (always available since nodejs 16.x)'
+        )
+    })
```
Comment (JadenSimon) [inline src/test/techdebt.test.ts:34]: We could start using `AbortController` too

**Review by JadenSimon — APPROVED** [review]

File: src/test/techdebt.test.ts (hunk 2 above)
Comment (JadenSimon) [reply src/test/techdebt.test.ts:34]: Kind of? `AbortController` becomes the 'official' way to cancel things related to Node. Like terminating a child process is a lot cleaner. Same for HTTP requests.

File: src/shared/vscode/env.ts (hunk 1 above)
Comment (justinmk3) [reply src/shared/vscode/env.ts:96]: based on https://docs.npmjs.com/cli/v7/configuring-npm/package-json#engines , `engines` is only advisory. So I'll skip it for now, since that would be an extra place to update.
